components/director/internal/domain/formation/service.go

```diff
@@ -128,23 +147,46 @@ func (s *service) GetFormationsForObject(ctx context.Context, tnt string, objTyp
 
 // CreateFormation adds the provided formation to the scenario label definitions of the given tenant.
 // If the scenario label definition does not exist it will be created
-func (s *service) CreateFormation(ctx context.Context, tnt string, formation model.Formation) (*model.Formation, error) {
-	f, err := s.modifyFormations(ctx, tnt, formation.Name, addFormation)
+func (s *service) CreateFormation(ctx context.Context, tnt string, formation model.Formation, templateName *string) (*model.Formation, error) {
+	formationName := formation.Name
+	f, err := s.modifyFormations(ctx, tnt, formationName, addFormation)
 	if err != nil {
-		if apperrors.IsNotFoundError(err) {
-			if err = s.labelDefService.CreateWithFormations(ctx, tnt, []string{formation.Name}); err != nil {
-				return nil, err
-			}
-			return &model.Formation{Name: formation.Name}, nil
+		if !apperrors.IsNotFoundError(err) {
+			return nil, err
 		}
+		if err = s.labelDefService.CreateWithFormations(ctx, tnt, []string{formationName}); err != nil {
+			return nil, err
+		}
+		// TODO:: Currently we need to support both mechanisms of formation creation/deletion(through label definitions and Formations entity) for backwards compatibility
+		if err = s.createFormation(ctx, tnt, *templateName, formationName); err != nil {
+			return nil, err
+		}
+		return &model.Formation{Name: formationName}, nil
```

Comment:
What about the other new `ID`, `TenantID`, `FormationTemplateID` fields? Should they be added to the GQL schema and/or resolver and service?

Reply:
I'm not sure it's in the scope of this task? We could have a quick discussion about that